File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int m=part.size();
        string str="";
        for(char ch:s){
            str+=ch;
            if(str.size()>=m && str.substr(str.size()-m)==part) str.erase(str.size()-m);
        }
        return str;
    }
};

File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.java
class Solution {
    public String removeOccurrences(String s, String part) {
        int m=part.length();
        StringBuilder str=new StringBuilder();
        for(char ch:s.toCharArray()){
            str.append(ch);
            if(str.length()>=m && str.substring(str.length()-m).equals(part)) str.delete(str.length()-m,str.length());
        }
        return str.toString();
    }
}